tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/MsTestIntegration.cs

```diff
@@ -266,15 +268,31 @@ internal static TestModule GetOrCreateTestModuleFromTestAssemblyInfo<TAsmInfo>(T
             objTestAssemblyInfo,
             key =>
             {
+                // assemblyName??? MsTest lies about this, if we check the usage it always use the Assembly.Location or similar...
+                // eg:
+                // https://github.com/microsoft/testfx/blob/5c829c1633fdca211c4a96a52a200f61ec85bd5c/src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs#L379
+                // https://github.com/microsoft/testfx/blob/5c829c1633fdca211c4a96a52a200f61ec85bd5c/src/Adapter/MSTest.TestAdapter/Discovery/TypeEnumerator.cs#L145
                 if (assemblyName is not null)
                 {
-                    assemblyName = AssemblyName.GetAssemblyName(assemblyName).Name ?? string.Empty;
-                }
-                else
-                {
-                    assemblyName = string.Empty;
+                    try
+                    {
+                        if (File.Exists(assemblyName))
+                        {
+                            assemblyName = AssemblyName.GetAssemblyName(assemblyName).Name ?? string.Empty;
+                        }
+                        else
+                        {
+                            assemblyName = new AssemblyName(assemblyName).Name ?? string.Empty;
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Common.Log.Warning(ex, "Error getting assembly name from {AssemblyName}", assemblyName);
+                        assemblyName = string.Empty;
+                    }
                 }
 
+                assemblyName ??= string.Empty;
```

Comment:
nit: I'd argue, if we're going to do this, remove the `?? string.Empty` from the above calls (and the `assemblyName = string.Empty` in the catch)

Reply:
true, done in [a0f8b25](https://github.com/DataDog/dd-trace-dotnet/pull/6336/commits/a0f8b25038817a959ef2cd07c9202205de3b259a)